src/dune/exe.ml

```diff
@@ -201,6 +212,31 @@ let build_and_link_many ~programs ~linkages ~promote ?link_args ?o_files cctx =
           if linkage = Linkage.js then
             link_js ~name ~cm_files ~promote cctx
           else
+            let link_time_code_gen =
+              if Linkage.is_plugin linkage then
+                Result.map link_time_code_gen
+                  ~f:(fun { Link_time_code_gen.to_link; force_linkall } ->
+                    let to_link =
+                      let rec depends_on lib0 lib =
+                        Lib_name.equal lib0 (Lib.name lib)
+                        || List.exists (Lib.requires_exn lib)
+                             ~f:(depends_on lib0)
+                      in
+                      List.filter_map to_link ~f:(function
+                        | Module _ as x -> Some x
+                        | Lib lib as x ->
+                          if
+                            List.exists embed_in_plugin_libraries
+                              ~f:(fun (_, lib0) -> depends_on lib0 lib)
+                          then
+                            Some x
+                          else
+                            None)
+                    in
+                    { Link_time_code_gen.to_link; force_linkall })
```

Comment:
This code filters out the library dependencies which are not inverse dependencies of the libraries listed in the `(embed_in_plugin_libraries ...)` field. The remaining libraries are linked into the plugin. I would appreciate a second pair of eyes on this code.

Reply:
One thing that isn't clear to me is:

if a library is mentioned in `(embed_in_plugin_libraries ..)`, do we guarantee in will be linked in the plugin?

This code seems to suggest we also require that the library be a (compile or link?) dependency of the executable. But the name of the field suggests that the libraries should be unconditionally linked.

Reply:
I updated the PR along the lines of your suggestion -- now the libraries listed on that field are unconditionally linked, no questions asked. It does not try to be clever, just simple to understand and predictable.